rust/sleeper_core/tests/df_object_store_bug_check.rs

```diff
@@ -82,19 +137,32 @@ async fn should_fail_on_datafusion_bug() -> Result<()> {
             ParquetReadOptions::default(),
         )
         .await?;
-    let mem_read = frame.show().await;
-
-    assert!(mem_read.is_err());
+    let mem_read = frame.clone().show().await;
+    assert!(mem_read.is_ok());
+    let expected = [
+        "+------+------+",
+        "| key1 | key2 |",
+        "+------+------+",
+        "| 1    | 11   |",
+        "| 2    | 12   |",
+        "| 3    | 13   |",
+        "| 4    | 44   |",
+        "| 5    | 55   |",
+        "| 6    | 66   |",
+        "+------+------+",
+    ];
+    assert_batches_neq!(expected, &frame.collect().await?);
```

Comment:
Can we assert what actually comes back here? We could explain what's wrong with it in a comment?